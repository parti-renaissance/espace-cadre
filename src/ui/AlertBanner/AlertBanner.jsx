import { Alert as MuiAlert } from '@mui/material'
import { styled } from '@mui/system'
import PropTypes from 'prop-types'

const Alert = styled(MuiAlert)(
  ({ theme }) => `
  color: ${theme.palette.statusError};
  background: ${theme.palette.backgroundError};
  border-radius: 8px;
  elevation: none;
  align-items: center;
  `
)

function AlertBanner({ severity, message }) {
  if (!message) {
    return null
  }

  return (
    <Alert elevation={0} variant="filled" severity={severity}>
      {typeof message === 'object' ? message.message : message}
    </Alert>
  )
}

export default AlertBanner

AlertBanner.propTypes = {
  severity: PropTypes.string.isRequired,
  message: PropTypes.oneOfType([
    PropTypes.string.isRequired,
    PropTypes.shape({
      message: PropTypes.string.isRequired,
    }),
  ]).isRequired,
}
