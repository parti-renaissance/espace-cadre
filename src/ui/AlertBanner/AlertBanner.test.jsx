import { render } from '@testing-library/react'
import AlertBanner from 'ui/AlertBanner/AlertBanner'

vi.mock('@mui/material', () => ({
  Alert: ({ children, ...rest }) => <div {...rest}>{children}</div>,
}))

describe('AlertBanner', () => {
  it('displays AlertBanner with string message', () => {
    const { container } = render(<AlertBanner severity="foo-severity" message="FOO" />)

    expect(container).toMatchSnapshot()
  })
  it('displays AlertBanner with object message', () => {
    const { container } = render(<AlertBanner severity="foo-severity" message={{ message: 'FOO' }} />)

    expect(container).toMatchSnapshot()
  })
})
