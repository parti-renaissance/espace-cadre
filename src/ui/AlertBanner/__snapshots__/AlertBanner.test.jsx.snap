// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlertBanner > displays AlertBanner with object message 1`] = `
<div>
  <div
    class="css-23lavf"
    elevation="0"
    severity="foo-severity"
    variant="filled"
  >
    FOO
  </div>
</div>
`;

exports[`AlertBanner > displays AlertBanner with string message 1`] = `
<div>
  <div
    class="css-23lavf"
    elevation="0"
    severity="foo-severity"
    variant="filled"
  >
    FOO
  </div>
</div>
`;

exports[`AlertBanner displays AlertBanner with object message 1`] = `
<div>
  <div
    class="css-23lavf"
    elevation="0"
    severity="foo-severity"
    variant="filled"
  >
    FOO
  </div>
</div>
`;

exports[`AlertBanner displays AlertBanner with string message 1`] = `
<div>
  <div
    class="css-23lavf"
    elevation="0"
    severity="foo-severity"
    variant="filled"
  >
    FOO
  </div>
</div>
`;
